portals/portals.c

```diff
@@ -107,11 +108,13 @@ int PtlInit(void)
 
 void PtlFini(void)
 {
-	struct linked_list *current = dev_list;
-	while (current->next != NULL) {

```

Comment:
Couldn't you simply change this to `current != NULL` instead of adding an enclosing `if`?